### ISSUE TITLE
Packages not loaded from defined location

### ISSUE THREAD

**Issue by mortelee** [OpenNTF/dojo-webpack-plugin#121]
In the dojoLoader configuration we define the following packages:

>       { name: 'lodash', location: 'xxx/libraries/lodash', main: 'lodash' },
>       { name: 'postal', location: 'xxx/libraries/postal', main: 'lib/postal' },
>       { name: 'postal.federation', location: 'xxx/libraries/postal.federation', main: 'lib/postal.federation' },
>       { name: 'postal.xframe', location: 'xxx/libraries/postal.xframe', main: 'lib/postal.xframe' },

Lodash is not added as a dependency in our package.json file. node_modules directory contains lodash.

When loading the application the postal.xframe library loads lodash from node_modules directory where it should load from 'xxx/libraries/lodash' (that is what we configured...).
When I add lodash as a dependency in package.jon, lodash is loaded from 'xxx/libraries/lodash'.

How can we we make sure that lodash is loaded from the location that is defined in the dojoLoader configuration?

**Comment by chuckdumont** [comment]
When modules can be resolved to more than one location, webpack just uses the first one it finds.  To exclude the lodash located in node_modules, you should be able to use a rule condition in webpack.config.js similar to:

```javascript
{
  test: /\.js$/,
  exclude: ["node_module/lodash"]
}
```
See https://webpack.js.org/configuration/module/#condition for details on using rule conditions.

**Comment by mortelee** [comment]
Thanks for your fast reply.

We've tried above proposal and also the one below
```
     {
        test: /\.js$/,
        exclude: function (modulePath) {
          var result = false;
          if (modulePath.match(/lodash/)) {
            result = /node_modules[\\/]lodash/.test(modulePath)
            console.log(modulePath, result);
          }
          return result;
        }
      }
```

In the console logging we see that it only tries to load lodash from the node_modules directory:
xxx\node_modules\lodash\lodash.js true

We expected that also the folder as configured in the dojoConfig would be logged. This is not the case...

When loading the application we still see that lodash is being loaded from node_modules directory.
What are we doing wrong?

**Comment by chuckdumont** [comment]
If you remove lodash from the node_modules directory, then do you see the logging for it the xxx/libraries directory?

**Comment by mortelee** [comment]
When removing lodash from the node_modules directory, lodash gets loaded from the location configured in the dojoConfig.

**Comment by mortelee** [comment]
Found a way to get it working, but I don't know if it is the way forward.
When adding resolve.alias to let lodash point to xxx\libraries\lodash i get it to work.

**Comment by chuckdumont** [comment]
Glad you could find a solution.

dojo-webpack-plugin registers a module resolver with webpack to resolve modules based on the dojo config.  The problem arises when webpack's native resolvers also find the module.  In this case, webpack will prefer the results from the native resolver, and exclude rules only seem to work for only for loaders.

I'll see if I can come up with a way to get webpack to give preference to the dojo resolver.

**Comment by chuckdumont** [comment]
@mortelee, please try release 2.3.5-0 on npm.  This should resolve the issue and you should no longer need the resolve.alias workaround.  Thanks.

**Comment by mortelee** [comment]
@chuckdumont , Just tried out version 2.3.5-0. It now works without resolve.alias. Big thanks.

**Comment by chuckdumont** [comment]
Issue is fixed in release 2.4.0.
